### DISCUSSION TITLE
How can I add action to Actions of the ContentItem and hide a certain action of the ContentItem,please?

### DISCUSSION THREAD

**Discussion by effapp** [OrchardCMS/OrchardCore#15760] [Q&A] [answered]
How can I add action to Actions of the ContentItem and hide a certain action of the ContentItem,please?

![image](https://github.com/OrchardCMS/OrchardCore/assets/51013754/ce67aa11-295d-457f-8118-4f87a8fe9ae6)

**Comment by douwinga** [comment]
You should be able to make a `ContentDisplayDriver` for `SummaryAdmin` views that targets `ActionsMenu` .

```csharp
public class MyAdditionalActionButtonDriver : ContentDisplayDriver
{
    public override IDisplayResult Display(ContentItem contentItem)
    {
        return Shape("MyAdditionalActionButton_SummaryAdmin__Button__Actions", new ContentItemViewModel(contentItem))
            .Location("SummaryAdmin", "ActionsMenu:10");
    }
}
```

And then just add the driver to your `Startup.cs` and add your view.

I am not sure the best way to hide an action button. Maybe a placements file?

**Reply by effapp** [reply]
Hi @douwinga ,

Where can I find all targets like ActionMenu , please?

**Reply by douwinga** [reply]
I don't think there is a place to find all of the targets. I just look at the views/drivers in the OC source and find something that I know does something similar to what I am trying to do.

**Comment by effapp** [comment]
> You should be able to make a `ContentDisplayDriver` for `SummaryAdmin` views that targets `ActionsMenu` .
> 
> ```cs
> public class MyAdditionalActionButtonDriver : ContentDisplayDriver
> {
>     public override IDisplayResult Display(ContentItem contentItem)
>     {
>         return Shape("MyAdditionalActionButton_SummaryAdmin__Button__Actions", new ContentItemViewModel(contentItem))
>             .Location("SummaryAdmin", "ActionsMenu:10");
>     }
> }
> ```
> 
> And then just add the driver to your `Startup.cs` and add your view.
> 
> I am not sure the best way to hide an action button. Maybe a placements file?

Hi @douwinga ,

These codes can run, but there is a question that my additional action present in ActionsMenu of any content type, I hope that my additional action only present in ActionsMenu of the specific content type, how shoud I do,please?

**Reply by douwinga** [reply, marked as answer]
You should be able to use `RenderWhen` to handle showing a button only for a specific content type. Something like this
```csharp
 return Shape("MyAdditionalActionButton_SummaryAdmin__Button__Actions", new ContentItemViewModel(contentItem))
     .Location("SummaryAdmin", "ActionsMenu:10")
     .RenderWhen(() => Task.FromResult(contentItem.ContentType == "MyContentType"));
```
